txscript/sigcache.go

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) 2015 The btcsuite developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package txscript
+
+import (
+	"bytes"
+	"crypto/rand"
+	"sync"
+
+	"github.com/btcsuite/btcd/btcec"
+	"github.com/btcsuite/btcd/wire"
+)
+
+// sigInfo represents an entry in the SigCache. Entries in the sigcache are a
+// 3-tuple: (sigCache, sig, pubKey).
+type sigInfo struct {
+	// TODO(roasbeef): Make this a string so it's hash func blocksize
+	// generic?
+	sigHash wire.ShaHash
+	sig     btcec.Signature
```

Comment:
Indeed! I forgot that the sig and pubkey structs themselves actually contain pointers. I've changed the `sigInfo` struct to include the serialized versions of `btcec.Signature` and `btcec.PublicKey`. 

Additionally, I've updated all new tests to serialize, then deserialize both the sig and pubkey before checking the `sigCache` for existence.